From 0323bf3814fd7de8532329be09df00121cccfcd5 Mon Sep 17 00:00:00 2001
From: dljames <dljames@chromium.org>
Date: Wed, 28 Jan 2026 14:57:04 -0800
Subject: [PATCH] [Vertical Tabs] Implement Remaining TabStripRegionView overrides

Implements the following functions in VerticalTabStripRegionView:
- IsTabStripEditable
- DisableTabStripEditingForTesting
- IsTabStripCloseable

Functions related to animations (IsAnimating, StopAnimating) have
default implementations and will be handled by test hooks in the future
and then removed from the API.

Tests are added as part of this change.

Change-Id: Ie9e58a1177b8a7bdbac86fb7ded543569f1c0418
Bug: 467710547, 467710617
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7517958
Reviewed-by: Eshwar Stalin <estalin@chromium.org>
Reviewed-by: Atharv Maan <atharvmaan@chromium.org>
Commit-Queue: Darryl James <dljames@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1576224}
---

--- a/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
+++ b/chrome/browser/ui/views/frame/vertical_tab_strip_region_view.cc
@@ -198,35 +198,31 @@ void VerticalTabStripRegionView::ResetTa
 }
 
 bool VerticalTabStripRegionView::IsTabStripEditable() const {
-  // TODO(crbug.com/467710547): This needs to consider the drag context. Wait
-  // until that is implemented before updating this function.
-  NOTIMPLEMENTED();
-  return tab_strip_editable_for_testing_;
+  return tab_strip_editable_for_testing_ &&
+         (!drag_handler_ ||
+          !drag_handler_->GetDragContext()->GetDragController());
 }
 
 void VerticalTabStripRegionView::DisableTabStripEditingForTesting() const {
-  // TODO(crbug.com/467710617): Implement this in VerticalTabStripView.
-  NOTIMPLEMENTED();
+  const_cast<VerticalTabStripRegionView*>(this)
+      ->tab_strip_editable_for_testing_ = false;
 }
 
 bool VerticalTabStripRegionView::IsTabStripCloseable() const {
-  // TODO(crbug.com/467710547): Return TabDragContext::IsTabStripCloseable once
-  // it exists.
-  NOTIMPLEMENTED();
-  return true;
+  if (!drag_handler_) {
+    return true;
+  }
+  TabDragController* drag_controller =
+      drag_handler_->GetDragContext()->GetDragController();
+  return !drag_controller || drag_controller->IsMovingLastTab();
 }
 
 bool VerticalTabStripRegionView::IsAnimating() const {
-  // TODO(crbug.com/467710547): Return if the view or drag context is animating
-  // something.
-  NOTIMPLEMENTED();
-  return true;
+  return false;
 }
 
 void VerticalTabStripRegionView::StopAnimating() {
-  // TODO(crbug.com/467710547): Stop any ongoing animation in the
-  // VerticalTabStripView.
-  NOTIMPLEMENTED();
+  // Do nothing.
 }
 
 void VerticalTabStripRegionView::UpdateLoadingAnimations(
